tests/src/python/test_qgsfeature.py

```diff
@@ -192,6 +212,80 @@ def test_setAttribute(self):
             feat.setAttribute(10, 5)
         self.assertTrue(feat.setAttribute(0, 5))
 
+        # Test different type of attributes
+        attributes = [
+            -9585674563452,
+            34.3,
+            False,
+            "QGIS",
+            QVariant("foo"),
+            QDate(2023, 1, 1),
+            QTime(12, 11, 10),
+            QDateTime(QDate(2020, 5, 6), QTime(8, 9, 10))
+        ]
+        self.assertEqual(feat.attributeCount(), 1)
+        for attribute in attributes:
+            self.assertTrue(feat.setAttribute(0, attribute))
+            self.assertEqual(feat.attribute(0), attribute)
+
+            feat.setAttribute(0, None)
+            if int(QT_VERSION_STR.split('.')[0]) >= 6:
+                self.assertIsNone(feat.attribute(0))
+            else:
+                self.assertTrue(feat.attribute(0).isNull())
```

Comment:
Qt 5 and Qt 6 do not handle "None" `QVariant` the same way. This has nothing to do with the change introduced by this MR. Do you think this is the right approach to test it?

Reply:
@ptitjano 

    self.assertEqual(feat.attribute(0), NULL)

should work in both versions

Reply:
Done.